contrib/asan/Make.user.asan

```diff
@@ -21,6 +21,3 @@ override JULIA_BUILD_MODE=debug
 
 # make ASAN consume less memory
 export ASAN_OPTIONS=detect_leaks=0:fast_unwind_on_malloc=0:allow_user_segv_handler=1:malloc_context_size=2
-
-# tell libblastrampoline to not use RTLD_DEEPBIND
-export LBT_USE_RTLD_DEEPBIND=0
```

Comment:
We can remove `LBT_USE_RTLD_DEEPBIND` here now. I didn't touch `ASAN_OPTIONS` above since "consume less memory" part might still be relevant.

Reply:
Yup, especially for running the test suite (if you do `make testall` the `Make.user` is sourced, although I'm not sure that's how CI runs tests).